packages/task/src/common/task-protocol.ts

```diff
@@ -22,9 +22,14 @@ export const TaskServer = Symbol('TaskServer');
 export const TaskClient = Symbol('TaskClient');
 
 export interface TaskConfiguration {
-    readonly type: string;
-    /** A label that uniquely identifies a task configuration */
+    /**
+     * Source of the task configuration.
+     * For a configured task, it is the name of the root folder, while for a provided task, it is the name of the provider.
+     */
+    readonly source: string;
```

Comment:
please mark internal fields (not visible in the file), e.g. prefix with one or two underscores

and revert source from README, a user don't need to specify it

Reply:
yes, but programmatically these extensions/plugins can use `_source` internal variable, then a user should not know about it, also some custom tasks can introduce `source` meaning something completely different.

VS Code API reserves only `type` property mandatory we should be aligned with it.